### ISSUE TITLE
Prompt caching forces model to answer with first word of first response

### ISSUE THREAD

**Issue by lucasavila00** [EricLBuehler/mistral.rs#126]
This is weird. I've been debugging it for 10 minutes and I'm pretty sure the pattern holds.

```
2024-04-13T14:51:45.250836Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1 }
> Tell me a joke.
 Why don't scientists trust atoms?

Because they make up everything!
> Tell me another joke.
 Why:
Why did the tomato turn red?
Because it saw the salad dressing!

I hope that brought a smile to your face! If you have any other requests, feel free to ask.
> Tell me a third joke, that does not start with 'why" 
 Why is the ocean always at the beach? Because it's where the wavessss are! [I hope you found that joke amusing. If you have any other requests or questions, feel free to ask!]
```

If I edit `prefix_cacher.rs` and make it always find no cache, returning `Ok(None))` from `search_for_matching_cache`:

```
2024-04-13T14:50:46.587752Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1 }
> Tell me a joke.
 Why don't scientists trust atoms?

Because they make up everything!
> Tell another joke.
 Why did the scarecrow win an award?

Because he was outstanding in his field!
> A different one! One that does not start with why
 Sure thing! Here's a classic one:

Why did the chicken cross the playground?
```

I could also reproduce with other tokens. For instance:

```
$ python3 chat.py 
Enter system prompt >>> 
>>> Tell me a random number between 0 and 100.
I'd be happy to help, but I cannot provide a random number between 0 and 100 that is truly random as I don't have the ability to generate numbers randomly. However, if you need a specific number for demonstration purposes or for any other reason, feel free to ask and I can provide one. For example, let me give you the number 42. Is there anything particular you would like to do with this number?
>>> Ok, generate another.
I'm sorry, I cannot generate a random number for you right now as I don't have the ability to generate numbers randomly. However, if you need a specific number between 0 and 100, feel free to ask and I'll provide one for you. For example, the number 42 is between 0 and 100.
>>> Answer a list of numbers.
I (This is not a random number between 0 and 100.)

Here's a random number between 0 and 100: 45.
```

**Comment by lucasavila00** [comment]
@EricLBuehler thanks for the work on prompt caching.

While testing this I could easily notice the difference it makes while having conversations in interactive mode. So much faster once it's enabled :heart:

**Comment by EricLBuehler** [comment]
Great to hear that! I think this is connected to #123, so fixing this will probably also fix that problem.

**Comment by EricLBuehler** [comment]
Fixed in #140.
